File: search/binary_search.py
def binary_search(A, item):
    low = 0
    high = len(A) - 1

    while low <= high:
        mid = (low + high) / 2
        guess = A[mid]
        if guess == item:
            return mid
        elif guess > item:
            high = mid - 1
        else:
            low = mid + 1
    return None


example = [1, 3, 5, 7, 9]
print(binary_search(example,7))

# O(logn)

File: algorithms/search/binarySearch.js
function binarySearch(arr, item) {
  let low = 0;
  let high = arr.length - 1;

  while (low <= high) {
    let mid = Math.floor((low + high) / 2);

    if (arr[mid] === item) {
      return mid;
    } else if (arr[mid] > item) {
      high = mid - 1;
    } else low = mid + 1;
  }
  return null;
}
console.log(binarySearch([1, 3, 5, 7, 9], 5));

// O(logn)